interfaces/devdoc/interlocked_hl_requirements.md

```diff
@@ -0,0 +1,157 @@
+# interlocked_hl requirements
+=============
+
+## Overview
+
+`interlocked_hl` is a collection of interlocked high level routines.
```

Comment:
>interlocked_hl [](start = 1, length = 14)

I assume this was meant to be deleted from common/devdoc instead of just copied here? Technically I think interlocked_hl should be in common once this sync module is ready, because then it will not depend on the platform at all. Maybe should just remove this copy and leave the MD where it was for now...